### DISCUSSION TITLE
Is there a way to get started with react-table in TypeScript?

### DISCUSSION THREAD

**Discussion by cseas** [TanStack/table#2766] [Q&A] [answered]
I tried going through the [Quick Start](https://react-table.tanstack.com/docs/quick-start) tutorial and as expected I started getting type errors soon as I started the example.

From what I've gathered from the previous Issues and Discussions on this topic, TypeScript isn't going to have a stable support for react-table anytime soon. And as observed in #2664, getting started in react-table with TS is confusing enough so it's probably not something to put to production in a TS project.

The only working example I've found so far on the internet is [this one](https://github.com/ggascoigne/react-table-example) but it's too complex to understand when just getting started.

Are there any TypeScript examples available by the community that can help get started with react-table? Or are there any other alternative libraries that could be better suited to a TypeScript project?

**Comment by calumpeak** [comment]
That's honestly the best example I've found and it covers most use cases. The other thing I found was on DT - update your global TD's with this. It's not ideal, but it work and is a start https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-table#example-type-file

**Comment by cseas** [comment, marked as answer]
I was able to figure out which types to add once I stumbled onto [this page in the documentation](https://react-table.tanstack.com/docs/api/useTable).

Here's the quickstart example in typescript, might help beginners new to react-table: 
https://github.com/cseas/react-table-custom/blob/main/src/Table/Table.tsx
